test/utils/rfq_blocked_address_utils_test.ts

```diff
@@ -0,0 +1,83 @@
+import { expect } from '@0x/contracts-test-utils';
+import 'mocha';
+import { Connection } from 'typeorm';
+
+import { getDBConnectionAsync } from '../../src/db_connection';
+import { BlockedAddressEntity } from '../../src/entities/BlockedAddressEntity';
+import { RfqBlockedAddressUtils } from '../../src/utils/rfq_blocked_address_utils';
+
+import { setupDependenciesAsync, teardownDependenciesAsync } from './deployment';
+
+// Force reload of the app avoid variables being polluted between test suites
+delete require.cache[require.resolve('../../src/app')];
+
+const SUITE_NAME = 'rfqBlockedAddressUtils';
+const ttlMs = 50;
+
+describe(SUITE_NAME, () => {
+    let connection: Connection;
+    let rfqBlacklistUtils: RfqBlockedAddressUtils;
+
+    before(async () => {
+        await setupDependenciesAsync(SUITE_NAME);
+        connection = await getDBConnectionAsync();
+        await connection.synchronize(true);
+        rfqBlacklistUtils = new RfqBlockedAddressUtils(connection, new Set(), ttlMs);
+    });
+
+    after(async () => {
+        // reset DB
+        connection = await getDBConnectionAsync();
+        await connection.synchronize(true);
+        await teardownDependenciesAsync(SUITE_NAME);
+    });
+
+    beforeEach(async () => {
+        // reset DB
+        connection = await getDBConnectionAsync();
+        await connection.synchronize(true);
+        rfqBlacklistUtils = new RfqBlockedAddressUtils(connection, new Set(), ttlMs);
+    });
+
+    it('should use stale values via isBlocked', async () => {
+        const sampleBadAddress = '0xA10612Ee5432B6395d1F0d6fB2601299a1c64274';
+        expect(rfqBlacklistUtils.isBlocked(sampleBadAddress)).to.be.false();
+
+        // Add it to the blocked list
+        await connection.getRepository(BlockedAddressEntity).save({
+            address: sampleBadAddress.toLowerCase(),
+        });
+
+        expect(rfqBlacklistUtils.isBlocked(sampleBadAddress)).to.be.false();
```

Comment:
could add another check to wait TTL and check `isBlocked` again, should be true

Reply:
so there isn't a guarantee that the TTL will be enough for the DB update to happen. What we would need is to `await` on the promise of the DB update (which is currently private).

That said, waiting something like 1 second would pretty much guarantee the DB update finishes - but it in theory is a potential race condition.

Alternatively, could expose the `updatePromise` and await on it. Will noodle on this some more

Reply:
found something similar to exposing the `updatePromise`, but slightly better